atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraMultinodeEteTest.java

```diff
@@ -0,0 +1,237 @@
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.After;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.jayway.awaitility.Awaitility;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.cassandra.ImmutableCassandraCredentialsConfig;
+import com.palantir.atlasdb.cassandra.ImmutableCassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.http.AtlasDbHttpClients;
+import com.palantir.atlasdb.keyvalue.cassandra.CassandraClientFactory;
+import com.palantir.atlasdb.todo.ImmutableTodo;
+import com.palantir.atlasdb.todo.Todo;
+import com.palantir.atlasdb.todo.TodoResource;
+import com.palantir.docker.compose.DockerComposeRule;
+import com.palantir.docker.compose.connection.Container;
+import com.palantir.docker.compose.connection.DockerPort;
+import com.palantir.docker.compose.connection.waiting.HealthCheck;
+import com.palantir.docker.compose.connection.waiting.SuccessOrFailure;
+import com.palantir.docker.compose.execution.DockerComposeExecArgument;
+import com.palantir.docker.compose.execution.DockerComposeExecOption;
+import com.palantir.docker.compose.execution.DockerComposeRunArgument;
+import com.palantir.docker.compose.execution.DockerComposeRunOption;
+
+public class CassandraMultinodeEteTest {
+
+    private static final Gradle GRADLE_PREPARE_TASK =
+            Gradle.ensureTaskHasRun(":atlasdb-ete-test-utils:buildCassandraImage");
+
+    private static final int THRIFT_PORT_NUMBER_1 = 9160;
+    private static final int THRIFT_PORT_NUMBER_2 = 9161;
+    private static final int THRIFT_PORT_NUMBER_3 = 9162;
+
+    private static final int TIMELOCK_SERVER_PORT = 3828;
+
+    private static final String CONTAINER_LOGS_DIRECTORY = "container-logs/cassandra-multinode";
+
+    private static final List<String> CASSANDRA_NODES = ImmutableList.of("cassandra1", "cassandra2", "cassandra3");
+
+    private static final long MAX_CASSANDRA_NODE_DOWN_MILLIS = 30000;
+    private static final long MAX_CASSANDRA_NODES_RUNNING_MILLIS = 3000;
+
+    private static final DockerComposeRule CASSANDRA_DOCKER_SETUP = DockerComposeRule.builder()
+            .file("docker-compose.multiple-cassandra.yml")
+            .waitingForService("cassandra1", Container::areAllPortsOpen)
+            .waitingForService("cassandra2", Container::areAllPortsOpen)
+            .waitingForService("cassandra3", Container::areAllPortsOpen)
+            .waitingForService("ete1", toBeReady())
+            .saveLogsTo(CONTAINER_LOGS_DIRECTORY)
+            .build();
+
+
+    @ClassRule
+    public static final RuleChain CASSANDRA_DOCKER_SET_UP = RuleChain
```

Comment:
Differentiating between `SET_UP` and `SETUP` is really nasty. Perhaps this one can be `PREPARED_DOCKER_SETUP` and the inner one can be `MULTINODE_CASSANDRA_SETUP`.